substrate/primitives/api/proc-macro/src/utils.rs

```diff
@@ -15,23 +15,24 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use crate::common::API_VERSION_ATTRIBUTE;
+use inflector::Inflector;
 use proc_macro2::{Span, TokenStream};
-
+use proc_macro_crate::{crate_name, FoundCrate};
+use quote::{format_ident, quote, ToTokens};
 use syn::{
 	parse_quote, spanned::Spanned, token::And, Attribute, Error, FnArg, GenericArgument, Ident,
 	ImplItem, ItemImpl, Pat, Path, PathArguments, Result, ReturnType, Signature, Type, TypePath,
 };
 
-use quote::{format_ident, quote};
-
-use proc_macro_crate::{crate_name, FoundCrate};
-
-use crate::common::API_VERSION_ATTRIBUTE;
-
-use inflector::Inflector;
-
 /// Generates the access to the `sc_client` crate.
 pub fn generate_crate_access() -> TokenStream {
+	if let Ok(FoundCrate::Name(name)) = crate_name(&"frame") {
+		let path = format!("{}::deps::{}", name, "sp_api");
```

Comment:
This crate is only available with the `runtime` feature. So maybe doing the `match crate_name("sp-api") {` code first?

Reply:
good point, we could move this block of code into the `Err` branch of the `match` statement, that could still be unnecessarily executed eventually though... I would rather feature flag the `if` condition